### ISSUE TITLE
Interface resolveType is not optional

### ISSUE THREAD

**Issue by Sytten** [graphql-nexus/nexus#823]
It should be possible to define:
```typescript
export const UserError = interfaceType({
  definition(t) {
    t.string('id')
  },
  name: 'UserError',
})
```
But I get an error:
```
Argument of type '{ definition(t: InterfaceDefinitionBlock<"UserError">): void; name: "UserError"; }' is not assignable to parameter of type 'NexusInterfaceTypeConfig<"UserError">'.
  Property 'resolveType' is missing in type '{ definition(t: InterfaceDefinitionBlock<"UserError">): void; name: "UserError"; }' but required in type '{ resolveType: AbstractTypeResolver<"UserError">; }'
```

**Comment by jasonkuhrt** [comment]
> It should be possible to define:

Are you using https://nexusjs.org/docs/guides/abstract-types#centralized-strategy-resolvetype?

Then this is expected I think.

**Comment by Sytten** [comment]
I mean the definition is:
```
Optionally provide a custom type resolver function. If one is not provided, the default implementation will first look for __typename, then fallback to calling isTypeOf on each implementing Object type.
```
I want to use the default implementation in this case :)

**Comment by jasonkuhrt** [comment]
@Sytten The JSDoc might be misleading. If that strategy is enabled, then this is expected.

**Comment by jasonkuhrt** [comment]
Yeah something seems fishy here.

**Comment by tgriesser** [comment]
> Optionally provide a custom type resolver function. If one is not provided, the default implementation will first look for __typename, then fallback to calling isTypeOf on each implementing Object type.

Yeah I think this doc might have been from before we switched `resolveType` to be the default. 

At the moment, I believe `resolveType` is required if `__typename` isn't explicitly defined on all `sourceType` for members of the interface.

I feel like the [local abstract settings](https://github.com/graphql-nexus/nexus/issues/623) would be really helpful here. @jasonkuhrt have you given much thought on how difficult you think this might be?

**Comment by jasonkuhrt** [comment]
> I feel like the local abstract settings would be really helpful here. @jasonkuhrt have you given much thought on how difficult you think this might be?

No thoughts up to now! I haven't had any sense on level-of-interest for it so far. Great to hear you think it would be useful. Do you have any input from others about it too?
